app/src/shared/web3/web3Provider.js

```diff
@@ -41,9 +49,45 @@ export const getWeb3 = (): StreamrWeb3 => {
     if (typeof ethereum !== 'undefined') {
         return new StreamrWeb3(ethereum)
     } else if (typeof web3 !== 'undefined') {
-        return new StreamrWeb3(web3.currentProvider)
+        return new StreamrWeb3(web3.currentProvider, true)
+    }
+    return new StreamrWeb3(false, true)
+}
+
+export const validateWeb3 = async (_web3: Web3): Web3 => {
+    if ((_web3.isLegacy && !window.web3) ||
+        (!_web3.isLegacy && !window.ethereum)) {
+        throw new Web3NotSupportedError()
+    }
+
+    // enable metamask
+    if (!_web3.isLegacy) {
+        try {
+            await ethereum.enable()
+        } catch (e) {
+            throw new Web3NotEnabledError()
+        }
     }
-    return new StreamrWeb3(false)
+
+    if (!_web3.currentProvider) {
+        throw new Web3NotEnabledError()
+    }
+
+    try {
+        // Check accounts
+        const accounts = await _web3.eth.getAccounts()
+
+        if (!Array.isArray(accounts) || accounts.length === 0) {
+            throw new WalletLockedError()
+        }
+    } catch (e) {
+        throw new WalletLockedError()
```

Comment:
Should probably at least log the error as a warning so it's easier to debug if something unexpected happens inside `await _web3.eth.getAccounts()`